website/docs/getting-started/basics.md

```diff
@@ -8,15 +8,34 @@ odo abstracts Kubernetes concepts into developer focussed terminology. In this d
 
 ### Concepts
 
-#### Component
+* Component
+  A component is a microservice. Each component can be run and deployed separately. Examples of components: Warehouse API Backend, Inventory API, Web Frontend, Payment Backend
+  > A component in odo is an equivalent of a deployment or deploymentconfig in Kubernetes. Creating a new component in odo will create a deployment in Kubernetes or for git and binary based component creation, it will create deploymentconfig.
```

Comment:
1. Why is this quoted with a `>`? Same question for everywhere else in the doc.
2. IMO, odo component is not an equivalent of Deployment or DeploymentConfig. By default, odo creates a bunch of Kubernetes resources for a component:
    - Deployment (which creates a ReplicaSet and Pods)
    - Service (a Kubernetes Service; not odo service)
    - Storage (emptyDir or PVC depending on user's preference setting)
    - probably more that I might have missed.
3. We don't need to go into this level of detail for git/binary components. Also, it doesn't create just DC but other things that I mentioned above.

Reply:
I understand that the quoted part explain how the concepts are implemented in Kubernetes.

But it is probably not a good place to talk about these implementations. This concept page should help the reader make a mental representation of the objects handled by odo, without other distraction.

Reply:
>Why is this quoted with a >? Same question for everywhere else in the doc.

1. I figured we would put the Kubernetes part in a quoted text like [Philippe has done in his blog](https://feloy.medium.com/odo-a-tool-to-simplify-development-on-kubernetes-ce0df34313b7).
I can add a note about it in the beginning.

>IMO, odo component is not an equivalent of Deployment or DeploymentConfig. By default, odo creates a bunch of Kubernetes resources for a component:

    Deployment (which creates a ReplicaSet and Pods)
    Service (a Kubernetes Service; not odo service)
    Storage (emptyDir or PVC depending on user's preference setting)
    probably more that I might have missed.

2. Okay, I can change that.

>We don't need to go into this level of detail for git/binary components. Also, it doesn't create just DC but other things that I mentioned above.

3. Noted.

Reply:
>But it is probably not a good place to talk about these implementations. This concept page should help the reader make a mental representation of the objects handled by odo, without other distraction.

Okay okay. I will remove the quoted texts. @feloy, let me know if you have something specific in mind about formatting this.

Reply:
> >I understand that the quoted part explain how the concepts are implemented in Kubernetes.
> >
> > But it is probably not a good place to talk about these implementations. This concept page should help the reader make a mental representation of the objects handled by odo, without other distraction.
> 
> Okay okay. I will remove the quoted texts. @feloy, let me know if you have something specific in mind about formatting this.

On a second thought, we mention early on in this page that we will be talking about these concepts and their Kubernetes equivalent, so it does make sense to me to put them here.

Reply:
> I figured we would put the Kubernetes part in a quoted text like [Philippe has done in his blog](https://feloy.medium.com/odo-a-tool-to-simplify-development-on-kubernetes-ce0df34313b7).
>        I can add a note about it in the beginning.

We don't want to introduce k8s lingo here.

> On a second thought, we mention early on in this page that we will be talking about these concepts and their Kubernetes equivalent, so it does make sense to me to put them here.

My bad. We should not talk about k8s equivalents in this doc. Can you remove that statement as well?

Reply:
Okay. Where else do we mention it then? This seemed like the right place to me to explain about the K8s equivalent of these concepts.

Reply:
We could limit talking about Kubernetes in the Command Reference and Architecture pages

Reply:
But I do think that talking about the K8s equivalent here will help get a better understanding of odo in general. We do not need to talk about the deep implementation level details, just the high-level ones to make a relation between odo terminology and K8s terminology.

I understand that it might defeat the purpose of abstracting K8s out of the docs, but if I was reading this doc as odo user, having this relation would help me better understand the docs. Or I guess we can make a better effort at making these concepts better understandable.

Reply:
> I understand that it might defeat the purpose of abstracting K8s out of the docs,

This is why we don't want to introduce k8s lingo so early.

> but if I was reading this doc as odo user, having this relation would help me better understand the docs

We do not want application developers (our target audience) to deal with k8s lingo. It's for application architects and whoever _already_ has knowledge of k8s. Majority of odo users are not expected to have knowledge of k8s.

> Or I guess we can make a better effort at making these concepts better understandable.

And that we'll be doing in Command Reference and Architecture sections. Maybe at some places in tutorials where it really makes sense. But Getting Started is definitely not the place to introduce k8s lingo.